packages/platform-ios/native_modules.rb

```diff
@@ -5,24 +5,23 @@
 #
 require 'pathname'
 
-def use_native_modules!(root = "..", config = nil)
+def use_native_modules!(config = nil)
   if (!config)
     json = []
 
-    # Make sure `react-native config` is ran from your project root
-    Dir.chdir(root) do
-      IO.popen("./node_modules/.bin/react-native config") do |data|
-        while line = data.gets
-          json << line
-        end
+    # @todo: Do not use Yarn here, but find path dynamically
+    IO.popen("yarn run --silent react-native config") do |data|
```

Comment:
I am most likely going to revert back to the original resolution mechanism we had, which was spawning `node` and looking for a location of a file. 

Yarn is nice, because it does that for us. If we could assume certain `npm` compatibility, I guess we would be able to use it as well.

Thoughts?

Certainly, hardcoded path to `node_modules/.bin` is not a go in a mono-repository structure (where CLI can be even installed globally and we shouldn't make any assumptions about that - it's one of the hidden architecture decisions to be run independently of the location of the project).

Reply:
https://github.com/react-native-community/cli/pull/384

Reply:
Thanks for linking some context! We will need to find a better alternative! Will look into it to remind myself about some of the constraints.

Reply:
@thymikee 

> Since @react-native-community/cli is not declared as template dependency, there is a possibility for Node to not be able to resolve it. It may happen because package managers can put the package inside react-native/node_modules/ behind a symlink (like npm does in some scenarios) which is invisible to require.resolve.

I believe we can address this by using `node` and `require.resolve` in the following manner:

- Call `path.dirname(require.resolve('react-native/package.json'))` to get path to React Native
- Use `require.resolve('@react-native-community/cli/..../index.js', { paths: [reactNativePath] })` to get path to the CLI from context of React Native where **we have it in package.json**

We use this mechanism already inside `resolveNodeModuleDir` and it works quite well. I believe we didn't have it at the time of that PR.

Reply:
That may work, definitely worth exploring.

Reply:
Works flawlessly and resolves to an absolute path to the CLI, including supporting symlinks (while in development).

I consider this "resolved"!